src/components/text-field/react/TextField.tsx

```diff
@@ -92,20 +104,60 @@ const DEFAULT_PROPS: Partial<TextFieldProps> = {
 
 interface IInputNativeProps {
     id?: string;
+    inputRef?: RefObject<HTMLInputElement> | RefObject<HTMLTextAreaElement>;
     isDisabled?: boolean;
     placeholder?: string;
     type?: TextFieldType;
     value: string;
+    rows: number;
+    lineHeight: number;
     setFocus(focus: boolean): void;
+    setRows(rows: number): void;
     onChange(value: string): void;
 }
 
 const renderInputNative = (props: IInputNativeProps): ReactElement => {
-    const { id, isDisabled, placeholder, type, value, setFocus, onChange, ...forwardedProps } = props;
+    const {
+        id,
+        isDisabled,
+        lineHeight,
+        placeholder,
+        type,
+        value,
+        setFocus,
+        onChange,
+        inputRef,
+        rows,
+        setRows,
+        ...forwardedProps
+    } = props;
     const onFocus = (): void => setFocus(true);
     const onBlur = (): void => setFocus(false);
 
-    const handleChange = (event: React.ChangeEvent): void => onChange(get(event, 'target.value'));
+    const handleChange = (event: React.ChangeEvent): void => {
+        if (type === TextFieldType.textarea) {
+            /**
+             * HEAD's UP! This part is a little bit tricky. The idea here is to only
+             * display the necessary rows on the textarea. In order to dynamically adjust
+             * the height on that field, we need to:
+             * 1. Set the current amount of rows to the minimum. That will make the scroll appear.
+             * 2. With that, we will have the `scrollHeight`, meaning the height of the container adjusted to the current content
+             * 3. With the scroll height, we can figure out how many rows we need to use by dividing the scroll height
+             * by the line height.
+             * 4. With that number, we can readjust the number of rows on the text area. We need to do that here, if we leave that to
+             * the state change through React, there are some scenarios (resize, hitting ENTER or BACKSPACE which add or remove lines)
+             * when we will not see the update and the rows will be resized to the minimum.
+             * 5. In case there is any other update on the component that changes the UI, we need to keep the number of rows
+             * on the state in order to allow React to re-render. Therefore, we save them using `useState`
+             */
+            (event.target as HTMLTextAreaElement).rows = MIN_ROWS;
+            const currentRows = event.target.scrollHeight / lineHeight;
+            (event.target as HTMLTextAreaElement).rows = currentRows;
+
+            setRows(currentRows);
+        }
+        onChange(get(event, 'target.value'));
+    };
```

Comment:
Could the compute of the number of rows be extracted in a separate hook that would depend on the textarea ref and produce the number of rows?
Something like:
```jsx
const rows = useComputeNumberOfRows(textAreaRef);
....
<textarea rows={rows}
```
The hook could be private to this component. It would use an internal `useState` to store the number of rows and a `useEffect` to listen to the on change event of the textarea (separatelly from this on change handler) to update the `rows` inner state

Reply:
Sure, great idea! I will take that approach :)